scripts/changelog.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+#
+# Copyright 2017 The rkt Authors
+# Copyright 2020 Kinvolk
+#
+# Create a changelog.
+#
+# The env variable RANGE specifies the range of commits to be searched for the changelog.
+# If unset the latest tag until origin/master will be set.
+#
+# The env variable GITHUB_TOKEN can specify a GitHub personal access token.
+# Otherwise one could run into GitHub rate limits. Go to
+# https://github.com/settings/tokens to generate a token.
+
+set -e
+
+jq --version >/dev/null 2>&1 || {
+    echo "could not find jq (JSON command line processor), is it installed?"
+    exit 255
+}
+
+if [ -z "${RANGE}" ]; then
+    LATEST_TAG=$(git describe --abbrev=0)
+    RANGE="${LATEST_TAG}..origin/master"
+fi
+
+if [ ! -z "${GITHUB_TOKEN}" ]; then
+    GITHUB_AUTH="--header \"authorization: Bearer ${GITHUB_TOKEN}\""
+fi
+
+for pr in $(git log --pretty=%s --first-parent "${RANGE}" | egrep -o '#\w+' | tr -d '#'); do
+    body=$(curl -s ${GITHUB_AUTH} https://api.github.com/repos/kinvolk/lokomotive/pulls/"${pr}" | \
+                  jq -r '{title: .title, body: .body}')
+
+    echo "-" \
+         "$(echo "${body}" | jq -r .title | sed 's/\.$//g')" \
+         "([#${pr}](https://github.com/kinvolk/lokomotive/pull/$pr))." \
+         "$(echo "${body}" | jq -r .body | awk -v RS='\r\n\r\n' NR==1 | tr -d '\r')"
```

Comment:
PR body can be rather lengthy sometimes, should be suggest including it in the changelog?

Reply:
Maybe we could pull the issues from GitHub milestone instead somehow?

Also, we could also consider using GitHub CLI:
```
$ gh pr list -B master -s merged

Pull requests for kinvolk/lokomotive

#157  docs: fix small typos and inconsistencies                                    iaguis/small-doc-fixes
#154  docs/packet: add ARM/x86 hybrid cluster instructions                         iaguis/add-arm-docs
#152  packet: use worker pool variable for the IPXE script URL                     iaguis/fix-ipxe-script-worker-pool
#148  Makefile: Only run linter if code compiles                                   rata/fix-linter-weird-errors
#146  packet: Use user-specified pool name for TF pool                             rata/packet-fixes
#143  Small fixes to bootkube and kube-apiserver                                   invidian/fixes
#141  lint: disable gochecknoglobals linter                                        iaguis/disable-global-vars-linter
#139  *: lokoctl {cluster,component} install -> lokoctl {cluster,component} apply  iaguis/apply
#138  examples/packet: add metallb_address_pool variable                           iaguis/fix-metallb-example
#136  cli/root: disable auto generated tag for CLI docs                            iaguis/disable-doc-tag
#135  Cleanup old documentation and adding backend configuration documentation.    imran/cleanup-old-documentation
#134  *: change quiet flag to verbose and disable it by default                    iaguis/verbose
#131  docs: add first how to guides                                                iaguis/howto-guides
#130  *: remove flannel and kube-router support                                    iaguis/remove-flannel
#125  Makefile: disable default excludes when running linter                       invidian/improve-linter
#121  docs: rework README and several other docs fixes                             iaguis/readme
#116  assets/lokomotive-kubernetes: fix formatting with 'terraform fmt'            invidian/fix-terraform-formatting
#114  contour: Update to v1.2.1                                                    surajssd/update-metallb
#113  metallb-dashboard: Add error and updates monitoring                          surajssd/install-metallb-dashboard
#112  pkg/terraform/executor.go: include key in Output error message               invidian/terraform-better-error-message
#111  Add Lokomotive concepts documentation                                        imran/lokomotive-concepts-documentation
#108  Improve handling kubeconfig path                                             invidian/kubeconfig-handling-improvement
#107  Small documentation improvements                                             invidian/docs-fixes
#106  Remove go-lint script and references in Makefile                             invidian/cleanup
#104  metallb: Add grafana dashboard                                               surajssd/add-metallb-dashboard
#103  Update all dependencies                                                      invidian/dep-updates
#102  Platform configuration guide for Bare Metal.                                 imran/platforms-configuration-guides-baremetal
#101  Changing heading from Installation to Installing.                            imran/installation-to-installing
#95   docs/packet: Add link on how to enable Local BGP                             rata:master
#94   Makefile: add 'all' and 'build-test' targets                                 invidian/makefile-all
```

But maybe later :)

Reply:
Yes, this would be nicer although what we're doing here is getting all merged PRs from a particular tag until master, and I'm not sure you can do that with the GitHub CLI...

Reply:
No worries, that was only a suggestion for improvements :)